staging/src/k8s.io/client-go/dynamic/client.go

```diff
@@ -66,6 +66,8 @@ type ResourceInterface interface {
 	DeleteCollection(deleteOptions *metav1.DeleteOptions, listOptions metav1.ListOptions) error
 	// Create creates the provided resource.
 	Create(obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
+	// CreateSubresource creates the provided resource.
+	CreateSubresource(name string, obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
```

Comment:
Could you remind me why CreateSubresource requires a name, instead of using obj.GetName()? And why we don't need an UpdateSubresource as well?

Reply:
We could use `obj.GetName()`, as it seems subresource name is implicitly always the same as the resource name. `CreateSubresource` has more code change because it's adding both `{name}` and `subresourceName` into the request:

`Create` posts to `.../resourceName`
`CreateSubresource` posts to `.../resourceName/{name}/subresourceName`
compared with other operations:
`Update` puts to `.../resourceName/{name}`
`UpdateSubresource` puts to `.../resourceName/{name}/subresourceName`

We have a separate `CreateSubresource` instead of merging the logic into `Create`, because we want the change to be more backward compatible.
(FYI, subresources that support create operation are: `pods/binding`, `pods/eviction` and some `deployments/rollback`)

Currently there are subresources supporting create/get/patch/update operations. Probably we don't need to change code in list/delete/deletecollection/watch operations.

Reply:
> Currently there are subresources supporting create/get/patch/update operations. Probably we don't need to change code in list/delete/deletecollection/watch operations.

I prefer not changing the methods that we are not supporting.

Reply:
Can you clarify why something like `CreateSubresource` is not a good idea?

Reply:
Having a separate `CreateSubresource` is for allowing user to input a resource {name} (`resourceName/{name}/subresourceName`) that is different from `obj.GetName()`. We think currently our system assumes every subresource object has the same name as the parent resource object. E.g. a pods/binding object having metadada.name "foo" means pod "foo" is being bound. So we can keep the dynamic client concise and restricted. 

cc @lavalamp